core/domain/blog_statistics_services.py

```diff
@@ -0,0 +1,1094 @@
+    blog_post_reads_aggregated_stats.reads_by_month = reads_stats.reads_by_month
+    blog_post_reads_aggregated_stats.reads_by_date = reads_stats.reads_by_date
+
+    blog_post_reads_aggregated_stats.update_timestamps()
+    blog_post_reads_aggregated_stats.put()
+
+
+def save_blog_post_reading_time_model(
+    reading_time_stats: blog_statistics_domain.BlogPostReadingTime
+) -> None:
+    """Updates the BlogPostReadingTimeModel datastore instance with the
+    passed BlogPostReadingTime domain object.
+
+    Args:
+        reading_time_stats: BlogPostReadingTime. The blog post reading time
+            stats domain object.
+
+    Raises:
+        Exception. No blog post reading time model exists for the given
+            blog_post_id.
+    """
+    reading_time_model = (
+        blog_stats_models.BlogPostReadingTimeModel.get(
+            reading_time_stats.blog_post_id, strict=False
+        )
+    )
+    if reading_time_model is None:
+        raise Exception(
+            'No blog post reading time model exists for the given blog_post_id.'
+        )
+
+    reading_time_model.zero_to_one_min = reading_time_stats.zero_to_one_min
+    reading_time_model.one_to_two_min = reading_time_stats.one_to_two_min
+    reading_time_model.two_to_three_min = reading_time_stats.two_to_three_min
+    reading_time_model.three_to_four_min = reading_time_stats.three_to_four_min
+    reading_time_model.four_to_five_min = reading_time_stats.four_to_five_min
+    reading_time_model.five_to_six_min = reading_time_stats.five_to_six_min
+    reading_time_model.six_to_seven_min = reading_time_stats.six_to_seven_min
+    reading_time_model.seven_to_eight_min = (
+        reading_time_stats.seven_to_eight_min
+    )
+    reading_time_model.eight_to_nine_min = reading_time_stats.eight_to_nine_min
+    reading_time_model.nine_to_ten_min = reading_time_stats.nine_to_ten_min
+    reading_time_model.more_than_ten_min = reading_time_stats.more_than_ten_min
+
+    reading_time_model.update_timestamps()
+    reading_time_model.put()
+
+
+def save_author_blog_post_views_stats_model(
+    views_stats: blog_statistics_domain.AuthorBlogPostViewsAggregatedStats
+) -> None:
+    """Updates the AuthorBlogPostViewsAggregatedStatsModel datastore instance
+    with the passed BlogPostViewsAggregatedStats domain object.
+
+    Args:
+        views_stats: AuthorBlogPostViewsAggregatedStats. The author blog post
+            views aggregated stats domain object.
+
+    Raises:
+        Exception. No author blog post views stats model exists for the given
+            author_id.
+    """
+    author_views_aggregated_stats = (
+        blog_stats_models.AuthorBlogPostViewsAggregatedStatsModel.get(
+            views_stats.author_id, strict=False
+        )
+    )
+    if author_views_aggregated_stats is None:
+        raise Exception(
+            'No author blog post views stats model exists for the given' +
+            ' author_id.'
+        )
```

Comment:
Same as above, please include author_id in the error message as well.

Reply:
Ditto below!

Reply:
I believe, we can not display author id to users due to data privacy. If we add it in error message here it will be reflected in frontend incase the error occurs.